src/store/backendApi.ts

```diff
@@ -43,6 +43,28 @@ export const useLazyGetOscapCustomizationsQuery = process.env.IS_ON_PREMISE
   ? cockpitQueries.useLazyGetOscapCustomizationsQuery
   : serviceQueries.useLazyGetOscapCustomizationsQuery;
 
+/**
+ * Hook to fetch OSCAP customizations for a specific compliance policy.
+ * Returns policy-specific customizations (packages, services, kernel args) that
+ * may differ from the base profile due to policy tailoring.
+ * 
+ * @returns undefined for on-premise environments (falls back to profile-based customizations)
+ */
+export const useGetOscapCustomizationsForPolicyQuery = process.env.IS_ON_PREMISE
+  ? undefined
```

Comment:
Since this is undefined, it might cause issues with the `on-prem` frontend. Maybe we could just use the same hook for both `on-prem` and the `hosted` service, but in the hook call itself we could add `process.env.IS_ON_PREMISE` as one of the `skip` conditions?